### PR TITLE
Status machine filter tests.

### PR DESCRIPTION
**Pull request by anastasiamac**
## Description of change

There are outdated bugs around status and its ability to filter by machine id.

This PR proposes tests to ensure that the functionality was delivered, is functioning as expected and that we do not regress in the future. 

These tests ensure that:
* we can filter by machine id to see units, and hence, applications that are relevant to this machine;
* we filter on machine id exactly, i.e. searching for machine "1" does not also return anything related to machine "10".

## Bug reference

https://bugs.launchpad.net/juju/+bug/1748915
https://bugs.launchpad.net/juju/+bug/1584170

### CODE REVIEW

**Comment by anastasiamac** [comment]
[status tests for filtering by machine number](https://trello.com/c/OdcIpBZW/258-status-tests-for-filtering-by-machine-number)

File: featuretests/cmd_juju_status_test.go [hunk 1]
```diff
@@ -20,9 +21,7 @@ type StatusSuite struct {
 	jujutesting.JujuConnSuite
 }
 
-func (s *StatusSuite) SetUpTest(c *gc.C) {
-	s.JujuConnSuite.SetUpTest(c)
-
+func (s *StatusSuite) assertMultipleRelationsBetweenApplications(c *gc.C) {
```
Comment (wallyworld) [inline featuretests/cmd_juju_status_test.go]: Maybe setupMultipleRelationsBetweenApplications
assertFoo() is typically used to test the results of doing something after it has happened and we want to check that the code behaved as expected.
Here, we are setting up a scenario for which we then want to run a test

File: featuretests/cmd_juju_status_test.go [hunk 2]
```diff
@@ -68,6 +67,7 @@ func (s *StatusSuite) run(c *gc.C, args ...string) *cmd.Context {
 }
 
 func (s *StatusSuite) TestMultipleRelationsInYamlFormat(c *gc.C) {
+	s.assertMultipleRelationsBetweenApplications(c)
```
Comment (wallyworld) [inline featuretests/cmd_juju_status_test.go]: As per above comment, this would be clearer as s.setupMultipleRelations.....

File: featuretests/cmd_juju_status_test.go [hunk 3]
```diff
@@ -92,10 +92,147 @@ func (s *StatusSuite) TestMultipleRelationsInYamlFormat(c *gc.C) {
...
 	context := s.run(c, "status", "--relations")
 	c.Assert(cmdtesting.Stdout(context), jc.Contains, `
 Relation provider      Requirer                   Interface  Type         Message
 wordpress:juju-info    logging:info               juju-info  subordinate  joining  
 wordpress:logging-dir  logging:logging-directory  logging    subordinate  joining  
 `[1:])
 }
+
+func (s *StatusSuite) assertServeralUnitsOnAMachines(c *gc.C) {
```
Comment (wallyworld) [inline featuretests/cmd_juju_status_test.go]: here too etc

File: featuretests/cmd_juju_status_test.go [hunk 4]
```diff
@@ -92,10 +92,147 @@ func (s *StatusSuite) TestMultipleRelationsInYamlFormat(c *gc.C) {
...
+		Application: applicationB,
+		Machine:     machine1,
+	})
+
+}
+
+func (s *StatusSuite) TestStatusWhenFilteringByMachine(c *gc.C) {
+	s.assertServeralUnitsOnAMachines(c)
+
+	// Put a unit from the application C on a different machine.
```
Comment (wallyworld) [inline featuretests/cmd_juju_status_test.go]: I didn't know what application C was until a dig in a bit. Maybe reword the comment just to say we are adding a new application to a machine ti ensure it is filtered out as expected.

File: featuretests/cmd_juju_status_test.go [hunk 5]
```diff
@@ -92,10 +92,147 @@ func (s *StatusSuite) TestMultipleRelationsInYamlFormat(c *gc.C) {
...
+	machine10 := s.Factory.MakeMachine(c, &factory.MachineParams{
+		Jobs:       []state.MachineJob{state.JobHostUnits},
+		InstanceId: instance.Id("id10"),
+	})
+
+	s.Factory.MakeUnit(c, &factory.UnitParams{
+		Application: application,
+		Machine:     machine10,
+	})
+	context := s.run(c, "status", "1")
```
Comment (wallyworld) [inline featuretests/cmd_juju_status_test.go]: Just to ensure and check there really is a machine 10, can we also run an check that "status 10" works as expected. Otherwise the test could pass by mistake.

**Comment by anastasiamac** [comment]
$$merge$$
